### PR TITLE
feat(trace-view): Link docs for missing services

### PR DESCRIPTION
**Pull request by wmak**
- This replaces an earlier attempt from https://github.com/getsentry/sentry/pull/25055
- This adds a new (???) node to quick trace when we think that there's
  probably a service that isn't being instrumented by Sentry
- This only shows up when the trace has exactly one transaction, and a
  platform recognized as backend or frontend
  - for Frontend services we show a (???) node in the cild spot
  - for Backend services we show a (???) node in the parent spot
- The dropdown has a dismiss that is permanent per user, we can probably
  change the key if we decide we want to start showing this again after
  EA or when we add more methods for determining missing services
  
## Preview:
- For an "angular" service
image
- For a ".NET" service
image

### CODE REVIEW

**Comment by silent1mezzo** [comment]
@wmak I'm concerned about allowing people to permanently dismiss this. There's a lot of value in connecting your frontend and backend projects. I wonder if dismissing with a TTL=1mo makes more sense?

**Comment by wmak** [comment]
@silent1mezzo addressed in 8737b44

File: src/sentry/static/sentry/app/components/quickTrace/index.tsx [hunk 1]
```diff
@@ -446,3 +494,77 @@ function StyledEventNode({text, hoverText, to, onClick, type = 'white'}: EventNo
...
+    return (
+      <React.Fragment>
+        {connectorSide === 'left' && <TraceConnector />}
+        <DropdownLink
+          caret={false}
+          title={<EventNode type="white">???</EventNode>}
+          anchorRight={anchor === 'right'}
+        >
+          <DropdownItem first width="small">
+            <ExternalDropdownLink href="https://docs.sentry.io/platforms/javascript/performance/connect-services/">
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/components/quickTrace/index.tsx:558]: Would analytics on this be interesting to have?

File: src/sentry/static/sentry/app/components/quickTrace/index.tsx [hunk 2]
```diff
@@ -151,6 +158,47 @@ export default function QuickTrace({
...
+  if (traceLength === 1) {
+    nodes.push(
+      <Projects
+        key="missing-services"
+        orgId={organization.slug}
+        slugs={[current.project_slug]}
+      >
+        {({projects}) => {
+          const project = projects.find(p => p.slug === current.project_slug);
+          if (project) {
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/components/quickTrace/index.tsx:170]: This way you won't need the optional chaining or type casting below.

```suggestion
          if (project?.platform) {
```

File: src/sentry/static/sentry/app/components/quickTrace/index.tsx [hunk 3]
```diff
@@ -151,6 +158,47 @@ export default function QuickTrace({
...
+            if (BACKEND_PLATFORMS.includes(project?.platform as string))
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/components/quickTrace/index.tsx:171]: Can we add in the `{}`?

**Review by Zylphrex — APPROVED** [review]

File: src/sentry/static/sentry/app/components/quickTrace/index.tsx (hunk 3 above)
Comment (wmak) [reply src/sentry/static/sentry/app/components/quickTrace/index.tsx:171]: @Zylphrex sorry not sure what you mean by add in the `{}`?

Comment (Zylphrex) [reply src/sentry/static/sentry/app/components/quickTrace/index.tsx]: You're missing the `{}` around the body of the `if` statement

Comment (wmak) [reply src/sentry/static/sentry/app/components/quickTrace/index.tsx]: 🤦 ah, yeah definitely can add it in, 🤔 thought the lint would've have opinions on it

**Review by silent1mezzo — APPROVED** [review]
👍

**Pull request merged**